### DISCUSSION TITLE
Handle notification messages if it originates from APNS

### DISCUSSION THREAD

**Discussion by yosus** [firebase/flutterfire#4951] [Feature request]
Hi,

With this v8-dev, on top of getting FCM token, I can also get the APNS token using `FirebaseMessaging.instance.getAPNSToken()`.

Both APNS and FCM tokens can be used to send notification to the iPhone device, but only notifications originating from FCM are triggering `onMessage `call. Those from APNS does not trigger anything.

Would love it if notifications originating from APNS can also trigger `onMessage`.

**Comment by alexandrefett** [comment]
Same for me. 
Acording FCM docs silent notifications are sent through APNs, not FCM servers.

_"On iOS, use this field to represent content-available in the APNs payload. When a notification or message is sent and this is set to true, **an inactive client app is awoken, and the message is sent through APNs as a silent notification and not through the FCM** connection server. Note that silent notifications in APNs are not guaranteed to be delivered, and can depend on factors such as the user turning on Low Power Mode, force quitting the app, etc."_
